pmd-java/src/main/java/net/sourceforge/pmd/lang/java/types/JTypeVar.java

```diff
@@ -112,17 +113,26 @@ default <T, P> T acceptVisitor(JTypeVisitor<T, P> visitor, P p) {
 
     /**
      * @throws UnsupportedOperationException If this is not a capture var
+     *
+     * <p>Note that this is only supposed to be used internally.
      */
+    @InternalApi
```

Comment:
Note, I'm not sure what to do about these. Should we deprecate them? They shouldn't be used outside our codebase, as not all implementations of the interface support those methods, they just have one use each in our codebase. Maybe they should be deprecated and converted to static methods in TypeOps...

Reply:
So they are making the build failed because japicmp thinks they were removed. It seems the only way to satisfy japicmp with our current config is to deprecate these methods, but if we mark them `@InternalApi`, it thinks they were removed...

Reply:
I think we need either an annotation like I describe [here](https://github.com/pmd/pmd/pull/4931#issuecomment-2040011418), to document that we will internalize these deprecated methods. Or we just document that in the `@deprecated` Javadoc tag, but we will need to be more careful when cleaning up for PMD 8.

Reply:
While we discuss this in #4931 I will revert these changes so that this PR is only about the bugfix. We can then open another PR to deprecate these methods.